webdriver/tests/bidi/browsing_context/locate_nodes/start_nodes.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget
+
+
+@pytest.mark.parametrize("type,value", [
+    ("css", "div"),
+    ("xpath", "//div"),
+    ("innerText", "foo")
+])
+@pytest.mark.asyncio
+async def test_locate_with_context_nodes(bidi_session, inline, top_context, type, value):
+    url = inline("""<p id="parent"><div data-class="one">foo</div><div data-class="two">foo</div></p>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    script_result = await bidi_session.script.evaluate(
+        expression="""document.querySelector("p")""",
+        target=ContextTarget(top_context["context"]),
+        await_promise=True,
+    )
+
+    parent_node_reference = { "sharedId": script_result["result"]["sharedId"] }
```

Comment:
Note that the returned value from `evaluate` is already the `result` so you can directly re-use it. Maybe rename `script_result` to `node_reference`.

Reply:
I _think_ I've got this right now? Please double-check my code.